motion_predict/test/test_predict_ctrv.cpp

```diff
@@ -139,26 +138,35 @@ TEST(predict_ctrv, predictStepExternal)
   obj.header.stamp = builtin_interfaces::msg::Time(rclcpp::Time(5, 0));
   obj.header.frame_id = "my_frame";
   obj.pose.pose.position.x = 5.0;
-  obj.pose.pose.orientation.w = 1.0;
+  obj.pose.pose.position.y = 1.4; // Added y position
+  // Set the orientation quaternion for 0.79 radians yaw
+  double yaw_angle = 0.79;
+  obj.pose.pose.orientation.w = cos(yaw_angle / 2);
+  obj.pose.pose.orientation.x = 0.0;
+  obj.pose.pose.orientation.y = 0.0;
+  obj.pose.pose.orientation.z = sin(yaw_angle / 2);
   obj.pose.covariance[0] = 1;
   obj.pose.covariance[7] = 1;
   obj.pose.covariance[35] = 1;
+  obj.velocity.twist.linear.x = 4.9244289009; // Initial velocity speed
   obj.velocity.covariance[0] = 999;
   obj.velocity.covariance[7] = 999;
   obj.velocity.covariance[35] = 999;
 
   carma_perception_msgs::msg::PredictedState result = motion_predict::ctrv::predictStep(obj, 0.1, 1000, 0.99);
 
-  ASSERT_NEAR(5.0, result.predicted_position.position.x, 0.00001);  // Verify update functions were called
-  ASSERT_NEAR(0.99, result.predicted_position_confidence, 0.01);
-  ASSERT_NEAR(0.001, result.predicted_velocity_confidence, 0.001);
+  EXPECT_NEAR(5.3466, result.predicted_position.position.x, 0.00001);  // Verify x position update
```

Comment:
Not a change request, but just realized that `EXPECT_*` is preferred over `ASSERT_*` by default (as you did here). I didn't know the difference before, but will follow this moving forward!

Reference from the [googletest primer](https://google.github.io/googletest/primer.html): "Usually EXPECT_* are preferred, as they allow more than one failure to be reported in a test. However, you should use ASSERT_* if it doesn’t make sense to continue when the assertion in question fails."

Reply:
awesome thanks. good to know. I only changed it so that I can see all the results at once instead of running it over and over